storagemarket/deal_execution.go

```diff
@@ -66,20 +82,30 @@ func (p *Provider) doDeal(deal *types.ProviderDealState, dh *dealHandler) {
 	// will fail, we can look into it when we implement deal completion.
 }
 
-func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter, deal *types.ProviderDealState, dh *dealHandler) error {
+func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter, deal *types.ProviderDealState, dh *dealHandler) *dealMakingError {
 	// publish "new deal" event
+	// TODO Do we need to fire this event for resumed deals ?
```

Comment:
@dirkmc Will the UI be able to handle this event getting fired again on a deal resumption on restart ?

Reply:
Yes it should do 👍

Reply:
removed this TODO